### PR TITLE
Add 'suppress' to messageable.send()

### PR DESCRIPTION
**Pull request by IKnewOne**
## Summary

Adds support for "suppress" argument for messageable.send(), mirroring message.edit() suppression argument.

## Information

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
- [x] I have updated the documentation to reflect the changes.
- [x] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Review by plun1331 — CHANGES_REQUESTED** [review]

File: discord/abc.py [hunk 1]
```diff
@@ -1401,6 +1407,8 @@ async def send(
             A list of stickers to upload. Must be a maximum of 3.
 
             .. versionadded:: 2.0
+        suppress: :class:`bool`
+            Indicates if message should suppress embeds.
```
Comment (Dorukyum) [inline discord/abc.py:1411]: Lib consistency
```suggestion
            Whether to suppress embeds for the message.
```

File: discord/abc.py [hunk 2]
```diff
@@ -1435,6 +1443,11 @@ async def send(
             if len(embeds) > 10:
                 raise InvalidArgument("embeds parameter must be a list of up to 10 elements")
             embeds = [embed.to_dict() for embed in embeds]
+            
+        if suppress is not None:
+            flags = MessageFlags.suppress_embeds
```
Comment (Dorukyum) [inline discord/abc.py:1448]: What if `suppress` is `False`?
```suggestion
        if suppress:
            flags = MessageFlags.suppress_embeds
```

File: discord/http.py [hunk 3]
```diff
@@ -449,6 +449,7 @@ def send_message(
         message_reference: Optional[message.MessageReference] = None,
         stickers: Optional[List[sticker.StickerItem]] = None,
         components: Optional[List[components.Component]] = None,
+        flags: Optional[int] = 0,
```
Comment (Dorukyum) [inline discord/http.py:452]: Why not use `None` if the parameter is optional
```suggestion
        flags: Optional[int] = None,
```

File: discord/http.py [hunk 4]
```diff
@@ -499,6 +503,7 @@ def send_multipart_helper(
...
```
Comment (Dorukyum) [inline discord/http.py:506]: Explanation above
```suggestion
        flags: Optional[int] = None,
```

File: discord/http.py [hunk 5]
```diff
@@ -556,6 +563,7 @@ def send_files(
...
```
Comment (Dorukyum) [inline discord/http.py:566]: Explanation above
```suggestion
        flags: Optional[int] = None,
```

**Review by Dorukyum — CHANGES_REQUESTED** [review]

File: discord/abc.py [hunk 6]
```diff
@@ -1401,6 +1407,8 @@ async def send(
             A list of stickers to upload. Must be a maximum of 3.
 
             .. versionadded:: 2.0
+        suppress: :class:`bool`
+            Whether to suppress embeds for the message.
```
Comment (EmreTech) [inline discord/abc.py:1411]:
```suggestion
            Whether or not embeds should be suppressed for this message.
```

File: discord/abc.py [hunk 7]
```diff
@@ -1435,6 +1443,11 @@ async def send(
             if len(embeds) > 10:
                 raise InvalidArgument("embeds parameter must be a list of up to 10 elements")
             embeds = [embed.to_dict() for embed in embeds]
+            
+        if suppress:
+            flags = MessageFlags.suppress_embeds
+        else:
+            flags = MessageFlags.DEFAULT_VALUE
```
Comment (EmreTech) [inline discord/abc.py:1450]:
```suggestion
        flags = MessageFlags.suppress_embeds if suppress else MessageFlags.DEFAULT_VALUE
```

**Review by EmreTech — CHANGES_REQUESTED** [review]

File: discord/abc.py (hunk 6 above)
Comment (Dorukyum) [reply discord/abc.py]: I suggested that line to make this consistent with `Message.edit`, `PartialMessage.edit` and `Paginator.edit`

**Review by EmreTech — APPROVED** [review]

**Review by Dorukyum — APPROVED** [review]

**Review by ChickenDevs — APPROVED** [review]

**Comment by honzajavorek** [comment]
Got

```
...
await thread.send(content=content, suppress=True)
TypeError: Messageable.send() got an unexpected keyword argument 'suppress'
```

when trying to use this with a thread, using version 2.1.3. It's late night here, so just noting it down as a comment. I'll try to verify and reproduce on a minimal example when I'm fresh and have more time. If I'm successful, I'll get back filing a separate issue.

**Comment by plun1331** [comment]
> Got
> 
> ```
> ...
> await thread.send(content=content, suppress=True)
> TypeError: Messageable.send() got an unexpected keyword argument 'suppress'
> ```
> 
> when trying to use this with a thread, using version 2.1.3. It's late night here, so just noting it down as a comment. I'll try to verify and reproduce on a minimal example when I'm fresh and have more time. If I'm successful, I'll get back filing a separate issue.

This change isn't in 2.1.3

**Comment by honzajavorek** [comment]
Oh! That would explain it then. This probably saved me hours of debugging, thank you 😀 I first got confused by docs (using the latest docs, but not the latest code from git means I may see new features not actually present in code), then by the order of the commits in the main branch, which got me thinking that this must be released since 2.1.2:

Screenshot 2022-09-14 at 11 27 12

Since `.edit()` supports it, workaround is easy (first send, then edit with suppress), so no stress.

**Pull request merged**